02.calendar/calendar.rb

```diff
@@ -0,0 +1,85 @@
+require 'date'
+
+class Calendar
+  WEEK_DAYS =
+    { 'Sun' => 1,
+      'Mon' => 2,
+      'Tue' => 3,
+      'Wed' => 4,
+      'Thu' => 5,
+      'Fri' => 6,
+      'Sat' => 7 }.freeze
+
+  def initialize(year, month)
+    @year = if !year.nil?
+              year
+            else
+              Date.today.year.to_s # 引数yearはnil以外でstringで渡ってくるので、この値もstringに合わせる
+            end
+
+    @month = if !month.nil?
+               month
+             else
+               Date.today.month.to_s
+             end
```

Comment:
```suggestion
    @year = (year || Date.today.year).to_i
    @month = (month || Date.today.month).to_i
```

最終的に文字列になるとしても、数字を文字列として扱うといろいろトラブルが起こるのでやめた方がいいと思います。

Reply:
これは全てのケースにおいて数字を文字列にするなということでしょうか。またはご指摘している箇所はインスタンス変数の定義の箇所なので、定義するときに簡単に数値をstringにするのは良くない。（各メソッド内で必要に応じて変換するのはOK）というお話でしょうか

Reply:
@shunhamm 正確には両方違います。

オブジェクト指向を考える上で、Calendarというクラスが持つプロパティであるyearやmonthがあったならそれがstringだというのは意味的におかしいというのが理由です。

中身を何も知らない人がCalendarクラスのyearプロパティがあるという情報を見たらintegerであると想定するのが自然です。

内部の実装上の都合よりもそういった点を考えて設計すべきだからです。

Reply:
なるほど！値の意味を考えるとそのあるべき状態で定義するべき型があるのでそれにするべきで、実装上の都合はそれが生じたところでstringにするなりするべきだ、ということですね、理解できましたー！